project-loader/src/main/kotlin/JUnit.kt

```diff
@@ -0,0 +1,105 @@
+package de.itemis.mps.gradle.junit
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.dataformat.xml.annotation.*
+
+data class Skipped(@JacksonXmlProperty(isAttribute = true)
+                   val message: String,
+                   @field:JacksonXmlCData()
+                   @field:JacksonXmlText() val content: String)
+
+data class Error(@field:JacksonXmlProperty(isAttribute = true)
+                 val message: String,
+                 @field:JacksonXmlProperty(isAttribute = true)
+                 val type: String? = null,
+                 @field:JacksonXmlCData()
+                 @field:JacksonXmlText() val content: String)
+
+
+data class Failure(@field:JacksonXmlProperty(isAttribute = true)
+                   val message: String,
+                   @field:JacksonXmlProperty(isAttribute = true)
+                   val type: String? = null,
+                   @field:JacksonXmlCData()
+                   @field:JacksonXmlText() val content: String)
+
+data class SystemOut(@field:JacksonXmlText()
+                     @field:JacksonXmlCData()
+                     val content: String)
+
+data class SystemErr(@field:JacksonXmlText()
+                     @field:JacksonXmlCData()
+                     val content: String)
+
+
+data class Testcase(
+        @field:JacksonXmlProperty(isAttribute = true)
+        val name: String,
+        @field:JacksonXmlProperty(isAttribute = true)
+        val classname: String,
+        @field:JacksonXmlProperty(isAttribute = true)
+        val assertions: Int? = null,
+        @field:JacksonXmlProperty(isAttribute = true)
+        val status: String? = null,
+        @field:JacksonXmlProperty(isAttribute = true)
+        val time: String? = null,
+        @field:JacksonXmlElementWrapper(useWrapping = false)
+        @field:JacksonXmlProperty(localName = "skipped")
+        @field:JsonInclude(JsonInclude.Include.NON_NULL)
```

Comment:
I know it looks strange to apply a JSON annotation to a XML serialized class but the Jackson XML serializer will respect this annotation as well and omit the value in the XML file when it's null.